Remove no longer spec'd color-contrast() support

weinig opened the pull request:
#### cd77f75e6f7b8e708fa349e4b286bc790b9c1271
<pre>
Remove no longer spec&apos;d color-contrast() support
<a href="https://bugs.webkit.org/show_bug.cgi?id=278864">https://bugs.webkit.org/show_bug.cgi?id=278864</a>

Reviewed by Tim Nguyen.

`color-contrast()` has been removed from CSS Color 5. Fortunately,
we never turned it on by default, so we can just remove it.

There is a replacement planned in CSS Color 6 called `contrast-color()`,
but there is nothing to be shared.

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/color-contrast-001-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/color-contrast-001.html: Removed.
    - Remove remaining test case, all others had been previously removed.

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/system-color-consistency.html:
    - Fix error in title I noticed while grepping for &quot;color-contrast&quot;.

* LayoutTests/inspector/unit-tests/css-keyword-completions.html:
    - Remove &quot;color-contrast()&quot; from set of completions.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
    - Remove color-contrast preference.

* Source/WebCore/css/CSSValueKeywords.in:
    - Remove keywords only used by &quot;color-contrast()&quot;.

* Source/WebCore/css/parser/CSSParserContext.cpp:
* Source/WebCore/css/parser/CSSParserContext.h:
    - Remove setting for enabling color contrast.

* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
(WebCore::CSSPropertyParserHelpers::consumeAColorFunction):
(WebCore::CSSPropertyParserHelpers::selectFirstColorThatMeetsOrExceedsTargetContrast): Deleted.
(WebCore::CSSPropertyParserHelpers::selectFirstColorWithHighestContrast): Deleted.
(WebCore::CSSPropertyParserHelpers::consumeColorContrastFunction): Deleted.
(WebCore::CSSPropertyParserHelpers::consumeColorRawForLegacyColorContrast): Deleted.
    - Remove implementation of &quot;color-contrast()&quot;.

Canonical link: <a href="https://commits.webkit.org/282921@main">https://commits.webkit.org/282921@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3972b6e932ab9b8b8e3de67517afbbde14880ef4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/64657 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/44022 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/17253 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/68680 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/15263 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/66774 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/51793 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/15542 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/52001 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10532 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/67723 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/40732 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55946 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32623 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37397 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/13321 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/14136 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/57766 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59325 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/13651 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/70385 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/63899 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/8602 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/13155 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/59328 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/8636 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/56031 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/59508 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14269 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/7113 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/779 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/85667 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/39834 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15109 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/40911 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/42094 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/40655 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Please export the WPT changes :)

weinig committed:
Remove no longer spec'd color-contrast() support
https://bugs.webkit.org/show_bug.cgi?id=278864

Reviewed by Tim Nguyen.

`color-contrast()` has been removed from CSS Color 5. Fortunately,
we never turned it on by default, so we can just remove it.

There is a replacement planned in CSS Color 6 called `contrast-color()`,
but there is nothing to be shared.

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/color-contrast-001-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/color-contrast-001.html: Removed.
    - Remove remaining test case, all others had been previously removed.

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/system-color-consistency.html:
    - Fix error in title I noticed while grepping for "color-contrast".

* LayoutTests/inspector/unit-tests/css-keyword-completions.html:
    - Remove "color-contrast()" from set of completions.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
    - Remove color-contrast preference.

* Source/WebCore/css/CSSValueKeywords.in:
    - Remove keywords only used by "color-contrast()".

* Source/WebCore/css/parser/CSSParserContext.cpp:
* Source/WebCore/css/parser/CSSParserContext.h:
    - Remove setting for enabling color contrast.

* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
(WebCore::CSSPropertyParserHelpers::consumeAColorFunction):
(WebCore::CSSPropertyParserHelpers::selectFirstColorThatMeetsOrExceedsTargetContrast): Deleted.
(WebCore::CSSPropertyParserHelpers::selectFirstColorWithHighestContrast): Deleted.
(WebCore::CSSPropertyParserHelpers::consumeColorContrastFunction): Deleted.
(WebCore::CSSPropertyParserHelpers::consumeColorRawForLegacyColorContrast): Deleted.
    - Remove implementation of "color-contrast()".

Canonical link: https://commits.webkit.org/282921@main
